lib/random-util.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const RANDOM_POOL_SIZE = 8192;
+const RANDOM_POOL_REFRESH = 1024;
+
+let randomPool = Buffer.alloc(RANDOM_POOL_SIZE);
+let randomPoolIdx = RANDOM_POOL_SIZE;
+
+function onRandomBytes (err, buf) {
+  randomPending = false;
+  if (!err) {
+    randomPool = buf;
+    randomPoolIdx = 0;
+  }
+}
+
+let randomPending = true;
+crypto.randomBytes(RANDOM_POOL_SIZE, onRandomBytes);
+
+function randomFillSync (buffer, offset, size) {
+  if (RANDOM_POOL_SIZE - randomPoolIdx < size) {
```

Comment:
What happens if randomFillSync is constantly called with size > RANDOM_POOL_SIZE we end up always calling crypto.randomFillSync that we like to avoid
 A better strategy IMHO will be to use the random pool as a ring buffer and try to refresh it with real randomness but if there is a need (we wrapped around the ring buffer) we do a quick XOR over the whole buffer with some math.rand  based number and fulfil the request and then try to refresh the random poll

Reply:
I update PR. It's always called with size === 4